### PR TITLE
[PROD-37308] Preserve original databricks.sdk.runtime for internal purposes

### PR DESCRIPTION
**Pull request by dby-tmwctw**
## Changes
We want to use `databricks-sdk` in DBR but currently installing it breaks existing `databricks.sdk.runtime` module we added. This PR fix it by using the internal packages if existed, and fall-back to OSS implementation if that does not exist.

## Tests
- Manual testing on E2 Dogfood

- [ ] `make test` run locally
- [x] `make fmt` applied
- [ ] relevant integration tests applied

### CODE REVIEW

File: databricks/sdk/runtime/__init__.py [hunk 1]
```diff
@@ -6,19 +6,42 @@ def _get_global_dbutils():
         return None
 
 
-dbutils = _get_global_dbutils()
+is_oss_implementation = True
```
Comment (nfx) [inline databricks/sdk/runtime/__init__.py:9]: i think a better name could be “local implementation”

File: databricks/sdk/runtime/__init__.py [hunk 2]
```diff
@@ -6,19 +6,42 @@ def _get_global_dbutils():
...
-        # this assumes that all environment variables are set
-        dbutils = RemoteDbUtils()
-except NameError:
-    from databricks.sdk.dbutils import RemoteDbUtils
-    if not dbutils:
-        # this assumes that all environment variables are set
-        dbutils = RemoteDbUtils()
+    # OSS implementation
+    is_oss_implementation = True
+    dbutils = _get_global_dbutils()
```
Comment (nfx) [inline databricks/sdk/runtime/__init__.py:30]: Can you check if we can remove _get_global_dbutils function altogether? It’s meant to check if “dbutils” got injected into nonlocal namespace

File: databricks/sdk/runtime/__init__.py [hunk 3]
```diff
@@ -6,19 +6,42 @@ def _get_global_dbutils():
...
+
+    try:
+        from .stub import *
```
Comment (nfx) [inline databricks/sdk/runtime/__init__.py:33]: Stub is only there for VSCODE extension to show Autocompletion with documentation. It has no implementation, as you see.

@kartikgupta-db do you think we can remove stub and keep VSCode extension functioning properly?

File: databricks/sdk/runtime/__init__.py [hunk 4]
```diff
@@ -6,19 +6,42 @@ def _get_global_dbutils():
...
+    except ImportError:
+        from databricks.sdk.dbutils import RemoteDbUtils
+        if not dbutils:
+            # this assumes that all environment variables are set
+            dbutils = RemoteDbUtils()
+    except NameError:
+        from databricks.sdk.dbutils import RemoteDbUtils
```
Comment (nfx) [inline databricks/sdk/runtime/__init__.py:40]: Why do we need to import remote dbutils implementation twice?…

**Review by nfx — CHANGES_REQUESTED** [review]
Few comments

File: databricks/sdk/runtime/__init__.py (hunk 3 above)
Comment (kartikgupta-db) [reply databricks/sdk/runtime/__init__.py:33]: Not for now. We have started fully depending on the SDK for the stubs. So we need to keep them in.

File: databricks/sdk/runtime/__init__.py (hunk 2 above)
Comment (kartikgupta-db) [reply databricks/sdk/runtime/__init__.py:30]: +1. here you are explicitly injecting dbutils into the globals namespace. So we don't really need  to see if dbutils are injected when `is_oss_implementation = True`

File: databricks/sdk/runtime/__init__.py [hunk 5]
```diff
@@ -6,19 +6,42 @@ def _get_global_dbutils():
...
+        if not dbutils:
+            # this assumes that all environment variables are set
+            dbutils = RemoteDbUtils()
```
Comment (kartikgupta-db) [inline databricks/sdk/runtime/__init__.py:43]:
```suggestion
    except (ImportError, NameError):
        from databricks.sdk.dbutils import RemoteDbUtils
        if not dbutils:
            # this assumes that all environment variables are set
            dbutils = RemoteDbUtils()
```

File: databricks/sdk/runtime/__init__.py (hunk 3 above)
Comment (nfx) [reply databricks/sdk/runtime/__init__.py:33]: @kartikgupta-db can we somehow separate the stubs even further within the SDK?

File: databricks/sdk/runtime/__init__.py (hunk 4 above)
Comment (dby-tmwctw) [reply databricks/sdk/runtime/__init__.py:40]: @nfx This is the original OSS implementation, but the commit suggestion by @kartikgupta-db down seems to solve this issue.

File: databricks/sdk/runtime/__init__.py (hunk 1 above)
Comment (dby-tmwctw) [reply databricks/sdk/runtime/__init__.py:9]: Makes sense, let me change the name.

File: databricks/sdk/runtime/__init__.py (hunk 2 above)
Comment (dby-tmwctw) [reply databricks/sdk/runtime/__init__.py:30]: @nfx @kartikgupta-db Makes sense, may I know how to test whether it is OK to remove it though, is passing unit tests sufficient?

**Review by nfx — APPROVED** [review]
Looks good to me, let’s wait for @kartikgupta-db to approve from VSCode part

File: databricks/sdk/runtime/__init__.py (hunk 2 above)
Comment (nfx) [reply databricks/sdk/runtime/__init__.py]: We don’t have unit tests for this module yet.

File: databricks/sdk/runtime/__init__.py [hunk 6]
```diff
@@ -1,24 +1,33 @@
...
+    except (ImportError, NameError):
+        from databricks.sdk.dbutils import RemoteDbUtils
+
         # this assumes that all environment variables are set
         dbutils = RemoteDbUtils()
 
-__all__ = ['dbutils']
+__all__ = ['dbutils'] if is_local_implementation else [
+    "display", "displayHTML", "dbutils", "table", "sql", "udf", "getArgument", "sc", "sqlContext", "spark"
+]
```
Comment (xiaochen-db) [inline databricks/sdk/runtime/__init__.py:33]: nit: store this list as constant since it repeats twice

Comment (dby-tmwctw) [reply databricks/sdk/runtime/__init__.py]: Done.

**Review by nfx — APPROVED** [review]
lgtm

**Pull request merged**